client/components/data/query-wordads-settings/index.jsx

```diff
@@ -2,19 +2,32 @@
  * External dependencies
  */
 import React from 'react';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import { requestWordadsSettings } from 'calypso/state/wordads/settings/actions';
+import isJetpackSite from 'calypso/state/sites/selectors/is-jetpack-site';
+import { requestJetpackSettings } from 'calypso/state/jetpack/settings/actions';
+import isRequestingJetpackSettings from 'calypso/state/selectors/is-requesting-jetpack-settings';
 
 export default function QueryWordadsSettings( { siteId } ) {
 	const dispatch = useDispatch();
+	const isJetpack = useSelector( ( state ) => isJetpackSite( state, siteId ) );
+	const isFetchingSettings = useSelector( ( state ) =>
+		isRequestingJetpackSettings( state, siteId, null )
+	);
 
 	React.useEffect( () => {
 		dispatch( requestWordadsSettings( siteId ) );
-	}, [ dispatch, siteId ] );
+
+		// WordAds settings on Jetpack sites are not available on the WordAds API endpoint, so we
+		// fetch them from the site settings endpoints.
+		if ( isJetpack && ! isFetchingSettings ) {
+			dispatch( requestJetpackSettings( siteId, null ) );
+		}
+	}, [ dispatch, siteId, isJetpack ] );
```

Comment:
Should we add `isFetchingSettings` in the deps array?

Reply:
Nope, because we don't want this effect to run again when `isFetchingSettings` changes, only if the Jetpack settings for the given site have not been requested yet. Will add a comment clarifying this.

Reply:
https://github.com/Automattic/wp-calypso/pull/53736/commits/5caa38008d15ffb2ee94f58d641db6619c46be50